Userland/Libraries/LibWeb/HTML/BrowsingContext.cpp

```diff
@@ -83,9 +84,11 @@ JS::NonnullGCPtr<BrowsingContext> BrowsingContext::create_a_new_top_level_browsi
 // https://html.spec.whatwg.org/multipage/browsers.html#creating-a-new-browsing-context
 JS::NonnullGCPtr<BrowsingContext> BrowsingContext::create_a_new_browsing_context(Page& page, JS::GCPtr<DOM::Document> creator, JS::GCPtr<DOM::Element> embedder, BrowsingContextGroup&)
 {
+    String webdriver_handle = Crypto::generate_random_uuid().release_value_but_fixme_should_propagate_errors();
```

Comment:
Feels a bit wasteful that every browsing context will go through the work of creating a UUID that is unused during normal usage. Maybe WebDriverConnection should instead set this on the Page/BrowsingContext when `set_is_webdriver_active` is called?

Reply:
hmm, I think that makes sense but I expect generating UUID to be pretty cheap operation. Also constructing BC is required only for a new tab/iframes which should not be that frequent.